pym/bob/generators/VisualStudio.py

```diff
@@ -0,0 +1,320 @@
+        super().__init__("vs2019", "Generate Visual Studio 2019 solution")
+        self.parser.add_argument('--uuid', help="Set solution UUID")
+
+    def configure(self, package, argv):
+        super().configure(package, argv)
+        if self.args.uuid:
+            self.uuid = UUID(self.args.uuid)
+        else:
+            self.uuid = randomUuid()
+
+    def generate(self, extra):
+        super().generate()
+        extra = " ".join(quote(e) for e in extra)
+
+        # gather root paths
+        bobPwd = Path(os.getcwd())
+        if sys.platform == 'msys':
+            if os.getenv('WD') is None:
+                raise BuildError("Cannot create Visual Studio project for Windows! MSYS2 must be started by msys2_shell.cmd script!")
+            msysRoot = PureWindowsPath(os.getenv('WD')) / '..' / '..'
+            winPwd = PureWindowsPath(os.popen('pwd -W').read().strip())
+            winDestination = PureWindowsPath(os.popen('cygpath -w {}'.format(quote(self.destination))).read().strip())
+        else:
+            winPwd = bobPwd
+            winDestination = Path(self.destination)
+
+        baseBuildMe = str(msysRoot / "msys2_shell.cmd") + \
+                " -msys2 -defterm -no-start -use-full-path -where " + \
+                str(winPwd)
+
+        rootPackage = self.rootPackage.getName()
+        projects = {}
+        for scan in self.packages:
+            name = scan.package.getName()
+            #p = projects.setdefault(name, Project(winDestination.joinpath(name)))
+            p = projects.setdefault(name, Project(winPwd, sha1NsUuid(self.uuid, name), "/".join(scan.package.getStack())))
+            for i in scan.headers: p.addHeader(PureWindowsPath(i))
+            for i in scan.sources: p.addSource(PureWindowsPath(i))
+            for i in scan.resources: p.addResource(PureWindowsPath(i))
+            for i in scan.incPaths: p.addIncPath(PureWindowsPath(i))
+            for i in scan.dependencies: p.addDependency(i)
+
+        buildMe = os.path.join(self.destination, "buildme.sh")
+        buildMeCmd = baseBuildMe + " " + buildMe
+        if not self.args.update:
+            self.updateFile(buildMe, self.__generateBuildme(extra))
+
+        solutionProjectList = []
+        solutionProjectConfigs = []
+        for name,project in projects.items():
+            p = os.path.join(self.destination, name)
+            os.makedirs(p, exist_ok=True)
+            self.updateFile(os.path.join(p, name+".vcxproj"), project.generateProject(projects, buildMeCmd),
+                    encoding="utf-8", newline='\r\n')
+            self.updateFile(os.path.join(p, name+".vcxproj.filters"), project.generateFilters(),
+                    encoding="utf-8", newline='\r\n')
+
+            solutionProjectList.append(SOLUTION_PROJECT_TEMPLATE.format(NAME=name, GUID=str(project.uuid).upper()))
+            solutionProjectConfigs.append("\t\t{{{GUID}}}.Build|x86.ActiveCfg = Build|Win32".format(GUID=str(project.uuid).upper()))
```

Comment:
Just for the root-project we have to add "\t\t{{{GUID}}}.Build|x86.Build.0", so the "Projektmappe" get a default project, which will be built by clicking <build "Projektmappe">.

Reply:
also fixed.